test/LibManagement.cpp

```diff
@@ -167,6 +167,46 @@ TEST_CASE("Valid manifests processing", "[Plugin]") {
     }
 }
 
+TEST_CASE("Valid download caching", "[Plugin]") {
+    smce::Toolchain tc{SMCE_PATH};
+    REQUIRE(!tc.check_suitable_environment());
+
+    if (std::filesystem::exists(CACHED_DOWNLOADS_PATH)) {
+        [[maybe_unused]] std::error_code ec_1;
+        std::filesystem::remove_all(CACHED_DOWNLOADS_PATH, ec_1);
+    }
+
+    std::vector<std::string> uris = {"https://github.com/ERROPiX/ESP32_AnalogWrite/archive/refs/tags/0.2.zip",
+                                     "https://github.com/ERROPiX/ESP32_AnalogWrite/archive/refs/tags/0.1.zip",
+                                     "https://github.com/ERROPiX/ESP32_AnalogWrite/archive/refs/tags/0.2.zip"};
+
+    std::vector<std::string> versions = {"0.2", "0.1", "0.2"};
```

Comment:
Those could be `std::array<std::string_view, 3>`s

Reply:
wont that cause conflicts since PluginManifest version expects std::string?

Reply:
`std::string` has a ctor for types convertible to `std::string_view` (and it's definitely convertible to itself)

Reply:
i changed it, is that what we are looking for?

Reply:
Yes, although the explicit functional cast to `std::string` does look odd